app/grandchallenge/components/models.py

```diff
@@ -2156,14 +2278,43 @@ def create_civ_for_file(self, *, ci, current_civ, new_value):
                         "user_upload_pk": str(new_value.pk),
                         "interface_pk": str(ci.pk),
                         "object_pk": self.pk,
-                        "civ_pk": current_civ.pk if current_civ else None,
+                        "linked_task": linked_task,
                     }
                 ).apply_async
             )
         elif not new_value:
             # if no new value is provided (user selects '---' in dropdown)
             # delete old CIV
-            self.values.remove(current_civ)
+            self.remove_civ(current_civ)
+
+    def get_civ_for_interface(self, interface):
+        raise NotImplementedError
+
+    def get_current_value_for_interface(
+        self, *, interface, user_upload=None, upload_session=None
+    ):
+        try:
+            return self.get_civ_for_interface(interface=interface)
+        except ObjectDoesNotExist:
+            return None
+        except MultipleObjectsReturned as e:
```

Comment:
If `MultipleObjectsReturned` is raised, we already get a sentry error (I'm re-raising the error further below). `ObjectDoesNotExist` can technically happen for display sets and archive items because the user can delete those (I think we've had sentry errors related to that in the past). For jobs it should indeed not happen, I can add a check for the model  name here and raise the error if it happens for Jobs.